File: lib/services/parse_script.py
import re
import os

file_path = "c:/Users/user/OneDrive/Desktop/TEMAN FLUTTER TRIAL/teman_flutter_app_code/lib/services/firestore_service.dart"

with open(file_path, "r", encoding="utf-8") as f:
    text = f.read()

sections = re.split(r'// ===================== (.*?) =====================', text)
print("Sections found:")
for i in range(1, len(sections), 2):
    print(sections[i])

File: tool/parse_script.dart
import 'dart:io';

void main() async {
  final file = File('c:/Users/user/OneDrive/Desktop/TEMAN FLUTTER TRIAL/teman_flutter_app_code/lib/services/firestore_service.dart');
  final content = await file.readAsString();
  
  final regex = RegExp(r'// ===================== (.*?) =====================');
  final matches = regex.allMatches(content);
  
  for (final match in matches) {
    print('Category: ${match.group(1)}');
  }
}
